pkg/cloud/userfile/filetable/file_table_read_writer.go

```diff
@@ -926,17 +934,28 @@ func (f *FileToTableSystem) NewFileWriter(
 		return nil, err
 	}
 
-	// BACKUP must allow overwriting of files. Since userfile is backed by a SQL
-	// table with filename as a PK, this would cause a constraint violation if
-	// we did not delete the file and its contents before writing.
-	//
-	// NB: userfile upload will error out on the client side if a file with the
-	// same name already exists.
-	err = f.deleteFileWithoutTxn(ctx, filename, e.ie)
```

Comment:
hmm this is interesting, I guess we don't want to do this anymore in a non-mixed version cluster? Since backups become write only we should never encounter a constraint violation. I wonder if removing this introduces a regression to other writers to userfile. Apart from backup I think it's only EXPORT, and I can't see why it would need to overwrite files.

Reply:
I think so but we can't get rid of it just yet right? Because of the latest file? After that one then yeah I don't see why we'd have to do it in non-mixed

Reply:
You're right, lets's add a TODO because until then userfile won't truly be write-only. Not a huge deal but worth fixing.